File: hadoop/mapreduce/src/main/java/com/cycloneboy/bigdata/hadoop/mapreduce/partiton/MyPartitioner.java
package com.cycloneboy.bigdata.hadoop.mapreduce.partiton;

import com.cycloneboy.bigdata.hadoop.mapreduce.flow.FlowBean;
import org.apache.hadoop.io.Text;
import org.apache.hadoop.mapreduce.Partitioner;

/** Create by sl on 2019-10-22 20:58 */
public class MyPartitioner extends Partitioner<Text, FlowBean> {

  @Override
  public int getPartition(Text text, FlowBean flowBean, int numPartitions) {
    String phone = text.toString();

    switch (phone.substring(0, 3)) {
      case "135":
        return 0;
      case "136":
        return 1;
      case "137":
        return 2;
      case "138":
        return 3;
      default:
        return 4;
    }
  }
}

File: flink-learn/flink-base/src/main/scala/com/cycloneboy/bigdata/flink/base/chapter5/util/MyPartitioner.scala
package com.cycloneboy.bigdata.flink.base.chapter5.util

import org.apache.flink.api.common.functions.Partitioner

import scala.util.Random

/**
 *
 * Create by  sl on 2020-04-06 11:05
 */
object MyPartitioner extends Partitioner[Int] {

  val random: Random.type = scala.util.Random

  override def partition(key: Int, numPartitions: Int): Int = {

    if (key < 0) 0 else random.nextInt(numPartitions)
  }

}
